SPECS/node-problem-detector/001-remove_arch_specific_makefile_logic.patch

```diff
@@ -23,58 +31,90 @@ index 84caddc..20b14f1 100644
  ALL_TARBALLS = $(foreach platform, $(PLATFORMS), $(NPD_NAME_VERSION)-$(platform).tar.gz)
  
 -output/windows_amd64/bin/%.exe: $(PKG_SOURCES)
--	GOOS=windows GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on go build \
--		-mod vendor \
+-	GOOS=windows GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) go build \
 -		-o $@ \
 -		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
 -		-tags "$(WINDOWS_BUILD_TAGS)" \
 -		./cmd/$(subst -,,$*)
 -	touch $@
 -
 -output/windows_amd64/test/bin/%.exe: $(PKG_SOURCES)
--	GOOS=windows GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on go build \
--		-mod vendor \
--		-o $@ \
+-	cd test && \
+-	GOOS=windows GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) go build \
+-		-o ../$@ \
 -		-tags "$(WINDOWS_BUILD_TAGS)" \
--		./test/e2e/$(subst -,,$*)
+-		./e2e/$(subst -,,$*)
 -
 -output/linux_amd64/bin/%: $(PKG_SOURCES)
--	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
+-	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) \
 -	  CC=x86_64-linux-gnu-gcc go build \
--		-mod vendor \
--		-o $@ \
--		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
--		-tags "$(LINUX_BUILD_TAGS)" \
--		./cmd/$(subst -,,$*)
--	touch $@
--
--output/linux_amd64/test/bin/%: $(PKG_SOURCES)
--	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
--	  CC=x86_64-linux-gnu-gcc go build \
--		-mod vendor \
--		-o $@ \
--		-tags "$(LINUX_BUILD_TAGS)" \
--		./test/e2e/$(subst -,,$*)
- 
--output/linux_arm64/bin/%: $(PKG_SOURCES)
--	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
--	  CC=aarch64-linux-gnu-gcc go build \
 +output/linux/bin/%: $(PKG_SOURCES)
 +	GOOS=linux CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
 +	  go build \
- 		-mod vendor \
  		-o $@ \
  		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
-@@ -173,9 +139,9 @@ output/linux_arm64/bin/%: $(PKG_SOURCES)
+ 		-tags "$(LINUX_BUILD_TAGS)" \
  		./cmd/$(subst -,,$*)
  	touch $@
  
+-output/linux_amd64/test/bin/%: $(PKG_SOURCES)
+-	cd test && \
+-	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) \
+-	  CC=x86_64-linux-gnu-gcc go build \
+-		-o ../$@ \
+-		-tags "$(LINUX_BUILD_TAGS)" \
+-		./e2e/$(subst -,,$*)
+-
+-output/linux_arm64/bin/%: $(PKG_SOURCES)
+-	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) \
+-	  CC=aarch64-linux-gnu-gcc go build \
+-		-o $@ \
+-		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
+-		-tags "$(LINUX_BUILD_TAGS)" \
+-		./cmd/$(subst -,,$*)
+-	touch $@
+-
 -output/linux_arm64/test/bin/%: $(PKG_SOURCES)
--	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
+-	cd test && \
```

Comment:
Do we need to still cd into the test dir when running tests?

Reply:
Looks like the /test/bin path isn't being referenced via the build-binaries target anyway. I confirmed that the check section still functions when we run "make test" as well.